src/styles/components/typeahead.less

```diff
@@ -0,0 +1,108 @@
+.form-control.form-control-group {
+
+  .typeahead {
+
+     .form-control {
+
+       &.bootstrap-typeahead-input-hint {
+        // Overriding `!important`s in Canvas' .form-control-group
+        margin-bottom: 0 !important;
+        margin-top: 0 !important;
+      }
+    }
+
+    .dropdown-menu {
+      position: absolute;
+      width: 100%;
+
+      .dropdown-menu-list {
+
+        // These extends are all copying dropdown styles from reactjs-components
+        li {
+          &:extend(.clickable);
+          &:extend(.dropdown-menu-list li.is-selectable all);
+
+          &.active {
+            &:extend(.dropdown-menu-list li.is-selectable:hover);
+          }
+
+          .highlight {
+            font-size: inherit;
+            line-height: inherit;
+          }
+        }
+      }
+    }
+
+    &.inverse {
+
+      .bootstrap-typeahead-input {
+
+        .bootstrap-typeahead-input-hint {
+          color: color-lighten(@gray, 30);
+        }
+      }
+
+      // These extends are all copying dropdown styles from reactjs-components
+      .dropdown-menu {
+        &:extend(.dropdown-menu.inverse);
+
+        .dropdown-menu-list {
+          &:extend(.dropdown-menu-list.inverse);
+
+          li {
+            &:extend(.dropdown.open .dropdown-menu.inverse li.is-selectable all);
+
+            &.active {
+              &:extend(.dropdown.open .dropdown-menu.inverse li.is-selectable:hover);
+            }
+          }
+        }
+      }
+    }
+  }
+
+  &.form-control-inverse {
+
+    .typeahead {
+
+      .bootstrap-typeahead-input-main {
+
+        &:extend(.form-control-inverse all);
+        &:extend(.filter-input-text all);
+      }
+    }
+  }
+}
+
+@media (min-width: @screen-mini) {
+
+  .form-control-group.filter-input-text-group {
+
+    .typeahead {
+
+      .dropdown-menu {
+        margin-left: -32px;
+        // We are matching the width of .filter-input-text-group
+        // as defined in reactjs-components
+        min-width: 100px;
+      }
+    }
+  }
+}
+
+@media (min-width: @screen-small) {
+
+  .form-control-group.filter-input-text-group {
+
+    .typeahead {
+
+      .dropdown-menu {
+        margin-left: -33px;
+        // We are matching the width of .filter-input-text-group
+        // as defined in reactjs-components
+        min-width: 200px;
```

Comment:
Same as above, let's use the variables that these pixel values are derived from.